javatests/arcs/core/analysis/InformationFlowTest.kt

```diff
@@ -0,0 +1,124 @@
+package arcs.core.analysis
+
+import arcs.core.data.AccessPath
+import arcs.core.data.Check
+import arcs.core.data.Claim
+import arcs.core.data.InformationFlowLabel
+import arcs.core.data.InformationFlowLabel.Predicate
+import arcs.core.data.InformationFlowLabel.SemanticTag
+import arcs.core.data.Recipe
+import arcs.core.data.Recipe.Particle
+import arcs.core.data.proto.ManifestProto
+import arcs.core.data.proto.decodeRecipes
+import arcs.core.util.Log
+import arcs.repoutils.runfilesDir
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.Truth.assertWithMessage
+import java.io.File
+import org.junit.Assert.assertTrue
+import org.junit.Assert.assertNotNull
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class InformationFlowTest {
+    /** A helper function to decode a RecipeProto in a [file] in the testdata directory. */
+    private fun parseManifestWithSingleRecipe(file: String): Recipe {
+        val binPath =
+            runfilesDir() + "javatests/arcs/core/analysis/testdata/${file}.pb.bin"
+        val manifestProto = ManifestProto.parseFrom(File(binPath).readBytes())
+        val recipes = manifestProto.decodeRecipes()
+        return requireNotNull(recipes.firstOrNull())
+    }
+
+    private fun Check.asString(): String {
+        this as Check.Assert
+        return "${accessPath} is $predicate"
+    }
+
+    /**
+     * Describes a test for a DFA, where [file] contains the manifest proto, [ingresses] is the
+     * name of ingress handles, and [violation] is the expected violation (if any).
+     */
+    data class DFATest(
+        val file: String,
+        val ingresses: List<String>,
+        val violation: List<String> = emptyList()
+    )
+
+    /** Returns null if the recipe is valid. Otherwise, returns the first violating check. */
+    private fun getViolatingChecks(test: String, ingresses: List<String>): List<String> {
+        val recipe = parseManifestWithSingleRecipe(test)
+        val result = InformationFlow.computeLabels(recipe, ingresses)
+        return result.checks.flatMap { (particle, checks) ->
+            checks.filter {
+                check -> !result.verify(particle, check)
+            }.map { it.asString() }
+        }
+    }
+
+    @Test
+    fun successDFA() {
+        val tests = listOf(
+            DFATest("ok-directly-satisfied", listOf("P1")),
```

Comment:
Optional: you could put a magic comment in the file that indicates the ingress points, e.g. something like this:

```
//#ingress = P1
```

It wouldn't be hard to parse, and would let you inline this directly in the test file.

Reply:
Good idea. Let do this in a following PR? Doing this here would pull in all the test files as well.